### PR TITLE
Feature ui update

### PR DESCRIPTION
**Pull request by SheldonCheng**
Two main features
- Update jobs and talents listing to fetch from actual data (previously mock data)
- Add basic Toast component

### CODE REVIEW

File: ui/src/components/Menu.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  };
+
+  React.useEffect(() => {
+    window.addEventListener("click", handleWindowClick);
+    return () => {
+      window.removeEventListener("click", handleWindowClick);
+    };
+  }, []);
+
+  React.Children.only(children);
```
Comment (lxcid) [inline ui/src/components/Menu.tsx:75]: Nit: This verify there's only 1 children and returns it. maybe u want to put it in the JSX instead?

See https://reactjs.org/docs/react-api.html#reactchildrenonly

File: ui/src/components/Menu.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  const handleClick = (e: React.MouseEvent) => {
+    e.preventDefault();
+    if (e.currentTarget === btnRef.current) setActive((val) => !val);
+  };
+
+  return (
+    <Container>
+      <OverflowMenuButton onClick={handleClick} ref={btnRef} />
+      {active && children}
```
Comment (lxcid) [inline ui/src/components/Menu.tsx:85]: Like `{active && React.Children.only(children)}`

File: ui/src/screens/Jobs/hooks/useJobsReducer.ts [hunk 3]
```diff
@@ -119,8 +115,9 @@ export default function usePostsReducer(): [JobsState, JobsActions] {
   const id = match?.params?.id;
 
   React.useEffect(() => {
-    axios.get<JobsResponseData>(`${process.env.REACT_APP_API}/jobs`).then((res) => {
-      updateJobs(res.data.jobs);
+    axios.get<PostInterface[]>(`${process.env.REACT_APP_API}/jobs/1`).then((res) => {
```
Comment (lxcid) [inline ui/src/screens/Jobs/hooks/useJobsReducer.ts:118]: In #93, I introduce `JobsGoWhereApiClient` which is axios underneath but allow us to just pass relative URL + its authorisation token is send whenever possible.

File: ui/src/screens/Talents/hooks/useTalentsReducer.ts [hunk 4]
```diff
@@ -95,8 +91,8 @@ export default function useTalentsReducer(): [TalentsState, TalentsActions] {
   const id = match?.params.id;
 
   React.useEffect(() => {
-    axios.get<TalentsResponseData>(`${process.env.REACT_APP_API}/talents`).then((res) => {
-      updateTalents(res.data.talents);
+    axios.get<PostInterface[]>(`${process.env.REACT_APP_API}/talents/1`).then((res) => {
```
Comment (lxcid) [inline ui/src/screens/Talents/hooks/useTalentsReducer.ts:94]: In #93, I introduce `JobsGoWhereApiClient` which is axios underneath but allow us to just pass relative URL + its authorisation token is send whenever possible.

**Review by lxcid — APPROVED** [review]
LGTM! Most of my comments are for the future or nits.

File: ui/src/components/Menu.tsx (hunk 2 above)
Comment (SheldonCheng) [reply ui/src/components/Menu.tsx:85]: ah thanks for the tip!

**Comment by SheldonCheng** [comment]
~@lxcid Noted on your changes in #93 with `JobsGoWhereApiClient`. Maybe we merge that first then I update this branch to adopt that change?~
Oops this just got merged. I'll resolve #93 conflicts instead

**Pull request merged**